src/lib.rs

```diff
@@ -15,162 +15,210 @@
 // TODO
 // generic number instead of f64 for position?
 
-extern crate serialize;
+extern crate "rustc-serialize" as rustc_serialize;
 
-use std::collections::HashMap;
-use serialize::json::{self, ToJson};
+use std::collections::BTreeMap;
+use rustc_serialize::json::{self, ToJson, Json};
 
-
-/// ToGeoJson
-pub trait ToGeoJson {
-    // TODO: change this to return a generic GeoJSON object
-    fn to_geojson(&self) -> Geometry;
-}
+// /// ToGeoJson
+// pub trait ToGeoJson {
+//     // TODO: change this to return a generic GeoJSON object
+//     fn to_geojson(&self) -> Geometry;
+// }
 
 
 /// Pos (alias for Positions)
 ///
 /// [GeoJSON Format Specification § 2.1.1](http://geojson.org/geojson-spec.html#positions)
+#[derive(RustcEncodable, Clone)]
 pub struct Pos(pub Vec<f64>);
 
 impl ToJson for Pos {
     fn to_json(&self) -> json::Json {
-        let &Pos(ref nums) = self;
-        nums.to_json()
+        match self.clone() {
```

Comment:
that leads to a stack overflow, though the ref syntax you originally had does do the trick